selftests/functional/test_statuses.py

```diff
@@ -0,0 +1,206 @@
+                                            'test pre',
+                                            'test post',
+                                            'teardown pre']),
+                    'exception-setup-37d3': ('ERROR',
+                                             ['setup pre',
+                                              'teardown pre',
+                                              'teardown post']),
+                    'exception-test-8f7d': ('ERROR',
+                                            ['setup pre',
+                                             'setup post',
+                                             'test pre',
+                                             'teardown pre',
+                                             'teardown post']),
+                    'exception-teardown-dfb1': ('ERROR',
+                                                ['setup pre',
+                                                 'setup post',
+                                                 'test pre',
+                                                 'test post',
+                                                 'teardown pre']),
+                    'kill-test-57fe': ('ERROR',
+                                       ['setup pre',
+                                        'setup post',
+                                        'test pre']),
+                    }
+
+
+class TestStatuses(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+        test_file = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.path.pardir,
+                                                 ".data",
+                                                 'test_statuses.py'))
+        yaml_file = os.path.abspath(os.path.join(os.path.dirname(__file__),
+                                                 os.path.pardir,
+                                                 ".data",
+                                                 'test_statuses.yaml'))
+
+        cmd = ('%s run %s -m %s --sysinfo=off --job-results-dir %s --json -' %
+               (AVOCADO, test_file, yaml_file, self.tmpdir))
+
+        results = process.run(cmd, ignore_status=True)
+        self.results = json.loads(results.stdout)
+
+    def test(self):
+        missing_tests = []
+
+        # Testing each individual test results
+        for test in self.results['tests']:
+            variant = test['id'].split(';')[1]
+            expected = EXPECTED_RESULTS.get(variant, False)
+            if not expected:
+                missing_tests.append(variant)
+            else:
+                self._check_test(test, expected)
+
+        # Testing if all variants were covered
+        missing_msg = ' '.join(missing_tests)
+        self.assertEqual(missing_msg, '',
+                         "Expected results not found for variants: %s" %
+                         missing_msg)
```

Comment:
Well all kind of info is useful when something bad happens as it might be rare-to-reproduce bug. Anyway let's get this merged and we can improve it when that happens...